File: my_learning_and_research/another_website_langs/js/login_page.js
// Create the login page container
const body = document.body;
body.style.cssText = "display: flex; justify-content: center; align-items: center; height: 100vh; background-color: #f3f3f3; font-family: Arial, sans-serif;";

const container = document.createElement("div");
container.style.cssText = "background-color: white; padding: 2rem; border-radius: 8px; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2); text-align: center; width: 300px;";
body.appendChild(container);

// Title
const title = document.createElement("h1");
title.innerText = "Login";
container.appendChild(title);

// Username input
const usernameLabel = document.createElement("label");
usernameLabel.setAttribute("for", "username");
usernameLabel.innerText = "Username:";
container.appendChild(usernameLabel);

const usernameInput = document.createElement("input");
usernameInput.type = "text";
usernameInput.id = "username";
usernameInput.name = "username";
usernameInput.placeholder = "Enter your username";
usernameInput.style.cssText = "width: 100%; padding: 0.8rem; margin: 1rem 0; border: 1px solid #ccc; border-radius: 4px;";
container.appendChild(usernameInput);

// Password input
const passwordLabel = document.createElement("label");
passwordLabel.setAttribute("for", "password");
passwordLabel.innerText = "Password:";
container.appendChild(passwordLabel);

const passwordInput = document.createElement("input");
passwordInput.type = "password";
passwordInput.id = "password";
passwordInput.name = "password";
passwordInput.placeholder = "Enter your password";
passwordInput.style.cssText = "width: 100%; padding: 0.8rem; margin: 1rem 0; border: 1px solid #ccc; border-radius: 4px;";
container.appendChild(passwordInput);

// Login button
const loginButton = document.createElement("button");
loginButton.innerText = "Login";
loginButton.style.cssText = "background-color: #007bff; color: white; padding: 0.8rem 1.2rem; border: none; border-radius: 4px; cursor: pointer; width: 100%;";
container.appendChild(loginButton);

loginButton.addEventListener("click", () => {
  const username = usernameInput.value;
  const password = passwordInput.value;

  if (username && password) {
    alert(`Welcome, ${username}!`);
    showHomePage(username);
  } else {
    alert("Please enter both username and password.");
  }
});

// Home Page View
function showHomePage(username) {
  container.innerHTML = ""; // Clear the container

  const homeTitle = document.createElement("h1");
  homeTitle.innerText = "Welcome to the Home Page!";
  container.appendChild(homeTitle);

  const welcomeMessage = document.createElement("p");
  welcomeMessage.innerText = `You are successfully logged in as ${username}.`;
  container.appendChild(welcomeMessage);

  const logoutButton = document.createElement("a");
  logoutButton.innerText = "Logout";
  logoutButton.href = "#";
  logoutButton.style.cssText = "display: block; margin-top: 1rem; color: #007bff; text-decoration: none;";
  container.appendChild(logoutButton);

  logoutButton.addEventListener("click", () => {
    location.reload(); // Reload the page to show the login form
  });
}

File: my_learning_and_research/another_website_langs/go/login_page.go
package main

import (
	"fmt"
	"html/template"
	"net/http"
)

var tmpl = template.Must(template.New("main").Parse(`
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>{{ .Title }}</title>
  <style>
    * {
      margin: 0;
      padding: 0;
      box-sizing: border-box;
      font-family: 'Arial', sans-serif;
    }
    body {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100vh;
      background-color: #f3f3f3;
    }
    .container {
      background-color: white;
      padding: 2rem;
      border-radius: 8px;
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
      text-align: center;
      width: 300px;
    }
    input[type="text"], input[type="password"] {
      width: 100%;
      padding: 0.8rem;
      margin: 1rem 0;
      border: 1px solid #ccc;
      border-radius: 4px;
    }
    button {
      background-color: #007bff;
      color: white;
      padding: 0.8rem 1.2rem;
      border: none;
      border-radius: 4px;
      cursor: pointer;
      width: 100%;
    }
    button:hover {
      background-color: #0056b3;
    }
  </style>
</head>
<body>
  <div class="container">
    <h1>{{ .Message }}</h1>
    {{ if .LoginForm }}
    <form method="POST" action="/home">
      <label for="username">Username:</label>
      <input type="text" id="username" name="username" required placeholder="Enter your username">
      
      <label for="password">Password:</label>
      <input type="password" id="password" name="password" required placeholder="Enter your password">
      
      <button type="submit">Login</button>
    </form>
    {{ else }}
    <p>Welcome, {{ .Username }}!</p>
    <a href="/">Logout</a>
    {{ end }}
  </div>
</body>
</html>
`))

// Struct for passing data to the template
type PageData struct {
	Title     string
	Message   string
	LoginForm bool
	Username  string
}

func main() {
	http.HandleFunc("/", loginPage)
	http.HandleFunc("/home", homePage)

	fmt.Println("Server running on http://localhost:8080")
	http.ListenAndServe(":8080", nil)
}

func loginPage(w http.ResponseWriter, r *http.Request) {
	data := PageData{
		Title:     "Login Page",
		Message:   "Please Login",
		LoginForm: true,
	}
	tmpl.Execute(w, data)
}

func homePage(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		http.Redirect(w, r, "/", http.StatusSeeOther)
		return
	}

	username := r.FormValue("username")
	password := r.FormValue("password")

	// Basic validation (In real applications, validate securely)
	if username == "" || password == "" {
		http.Redirect(w, r, "/", http.StatusSeeOther)
		return
	}

	data := PageData{
		Title:     "Home Page",
		Message:   "Welcome to the Home Page!",
		LoginForm: false,
		Username:  username,
	}
	tmpl.Execute(w, data)
}